### PR TITLE
Add libroot.h and update rootless.h for rootless v2 

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
See title

Checklist
---------
- [x] New code follows the [code rules](https://github.com/theos/headers/blob/master/README.md#code-rules)
- [x] I've added modulemaps for any new libraries (e.g. see [libactivator/module.modulemap](https://github.com/theos/headers/blob/f3e596d896bae8f07c43cfb00ef55bf6224b4cdc/libactivator/module.modulemap)): it should be possible to `@import MyLibrary;` in ObjC, or `import MyLibrary` in Swift.
- [x] My contribution is code written by myself from reverse-engineered headers, licensed into the Public Domain as per [LICENSE.md](LICENSE.md); or, code written by myself / taken from an existing project, released under an OSI-approved license, and I've added relevant licensing credit to [LICENSE.md](LICENSE.md)

Does this close any currently open issues?
------------------------------------------
No

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
Tested working on Dopamine 2.0 (iP7, iOS 15.8). See commit message for more details. 
```
==> Compiling Tweak.x (arm64)…
Tweak.x:7:17: warning: 'ROOT_PATH_NS' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead. [-W#pragma-messages]
        NSString *l1 = ROOT_PATH_NS(@"my/test1");
                       ^
/home/lightmann/test/rootless.h:21:28: note: expanded from macro 'ROOT_PATH_NS'
#define ROOT_PATH_NS(path) _Pragma("message(\"'ROOT_PATH_NS' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead.\")") JBROOT_PATH_NSSTRING(path)
                           ^
<scratch space>:4:2: note: expanded from here
 message("'ROOT_PATH_NS' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead.")
 ^
Tweak.x:8:17: warning: 'ROOT_PATH_NS_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead. [-W#pragma-messages]
        NSString *l2 = ROOT_PATH_NS_VAR(@"my/test2");
                       ^
/home/lightmann/test/rootless.h:22:32: note: expanded from macro 'ROOT_PATH_NS_VAR'
#define ROOT_PATH_NS_VAR(path) _Pragma("message(\"'ROOT_PATH_NS_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead.\")") JBROOT_PATH_NSSTRING(path)
                               ^
<scratch space>:6:2: note: expanded from here
 message("'ROOT_PATH_NS_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_NSSTRING' instead.")
 ^
Tweak.x:9:19: warning: 'ROOT_PATH' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead. [-W#pragma-messages]
        const char *s3 = ROOT_PATH("my/test3");
                         ^
/home/lightmann/test/rootless.h:16:26: note: expanded from macro 'ROOT_PATH'
#define ROOT_PATH(cPath) _Pragma("message(\"'ROOT_PATH' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead.\")") JBROOT_PATH_CSTRING(cPath)
                         ^
<scratch space>:8:2: note: expanded from here
 message("'ROOT_PATH' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead.")
 ^
Tweak.x:10:19: warning: 'ROOT_PATH_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead. [-W#pragma-messages]
        const char *s4 = ROOT_PATH_VAR("my/test4");
                         ^
/home/lightmann/test/rootless.h:17:29: note: expanded from macro 'ROOT_PATH_VAR'
#define ROOT_PATH_VAR(path) _Pragma("message(\"'ROOT_PATH_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead.\")") JBROOT_PATH_CSTRING(path)
                            ^
<scratch space>:10:2: note: expanded from here
 message("'ROOT_PATH_VAR' is deprecated. Please import libroot.h and use 'JBROOT_PATH_CSTRING' instead.")
 ^
4 warnings generated.
```

Where has this been tested?
---------------------------
**Operating System:** …

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

File: rootless.h [hunk 1]
```diff
@@ -1,24 +1,22 @@
-#include <sys/syslimits.h>
-#include <unistd.h>
+// TODO: remove this file
```
Comment (leptos-null) [inline rootless.h:1]: Do we need any of the commented code? it's tracked in the git history and using `THEOS_PACKAGE_INSTALL_PREFIX` directly is not helpful is the vast majority of cases.

Comment (L1ghtmann) [reply rootless.h:1]: Nope, left them there for reference. Can remove them outright if desired or we can wait until we remove the file entirely in the future.

**Review by leptos-null — APPROVED** [review]
Excited about these changes.
This PR significantly simplifies `rootless.h` and exposes "rootless" API to Swift for the first time.

**Review by MrGcGamer — APPROVED** [review]
Looks good now, with the changes to allow usage from cpp

**Pull request merged**